src/key.h

```diff
@@ -203,6 +203,25 @@ class CKey
     ECDHSecret ComputeBIP324ECDHSecret(const EllSwiftPubKey& their_ellswift,
                                        const EllSwiftPubKey& our_ellswift,
                                        bool initiating) const;
+
+    /** Assumes the public key is compressed */
+    template <typename Stream>
+    inline void Unserialize(Stream& s) {
+        std::vector<unsigned char> key_data;
+        key_data.resize(32);
+        s >> MakeWritableByteSpan(key_data);
+        this->Set(key_data.begin(), key_data.end(), true);
+        memory_cleanse(key_data.data(), key_data.size());
```

Comment:
Use `secure_allocator` for the vector because it also ensures the data is not swapped out to disk. Then you do not need the `memory_cleanse()` call at the end:

```suggestion
        std::vector<unsigned char, secure_allocator<unsigned char>> key_data(32);
        s >> MakeWritableByteSpan(key_data);
        this->Set(key_data.begin(), key_data.end(), true);
```

Reply:
Wait, I forgot that this is inside a `CKey` method and we have access to the private `keydata` member. Should be possible to write directly to it, without an intermediate vector. Something like:

```cpp
MakeKeyData();
s >> MakeSpanFromCKeykeydataomgwhatisthis(keydata->begin(), keydata->end());
```